kafka-ui-api/src/main/java/com/provectus/kafka/ui/service/SchemaRegistryService.java

```diff
@@ -144,25 +146,25 @@ private SchemaSubjectDTO withSchemaType(SchemaSubjectDTO s) {
   private Mono<ResponseEntity<Void>> deleteSchemaSubject(KafkaCluster cluster, String schemaName,
                                                          String version) {
     return configuredWebClient(
-        cluster,
-        HttpMethod.DELETE,
-        URL_SUBJECT_BY_VERSION, schemaName, version)
-        .retrieve()
-        .onStatus(NOT_FOUND::equals,
-            throwIfNotFoundStatus(formatted(NO_SUCH_SCHEMA_VERSION, schemaName, version))
-        ).toBodilessEntity();
+            cluster,
+            HttpMethod.DELETE,
+            URL_SUBJECT_BY_VERSION, schemaName, version)
+            .retrieve()
+            .onStatus(NOT_FOUND::equals,
+                    errorOnStatusNotFound(formatted(NO_SUCH_SCHEMA_VERSION, schemaName, version))
+            ).toBodilessEntity();
   }
 
-  public Mono<ResponseEntity<Void>> deleteSchemaSubjectEntirely(KafkaCluster cluster,
-                                                                String schemaName) {
+  public Mono<Void> deleteSchemaSubjectEntirely(KafkaCluster cluster,
+                                                String schemaName) {
     return configuredWebClient(
-        cluster,
-        HttpMethod.DELETE,
-        URL_SUBJECT, schemaName)
-        .retrieve()
-        .onStatus(NOT_FOUND::equals,
-            throwIfNotFoundStatus(formatted(NO_SUCH_SCHEMA, schemaName)))
-        .toBodilessEntity();
+            cluster,
+            HttpMethod.DELETE,
+            URL_SUBJECT, schemaName)
+            .retrieve()
+            .onStatus(HttpStatus::isError, errorOnSchemaDeleteFailure(schemaName))
+            .toBodilessEntity()
+            .flatMap(r -> Mono.empty());
```

Comment:
use then() instead  of .flatMap(r -> Mono.empty()); Also apply this return type change to all deleteXXX methods